modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -602,10 +602,22 @@
            handle-limit
            handle-page]))
 
+(defn- query->collection-name
+  "Return `:collection` from a source query, if it exists."
+  [query]
+  (mbql.u/match-one query
+    (_ :guard (every-pred map? :collection))
+    ;; ignore source queries inside `:joins` or `:collection` outside of a `:source-query`
+    (when (and (= (last &parents) :source-query)
+               (not (contains? (set &parents) :joins)))
+      (:collection &match))))
+
```

Comment:
Mongo stuff is not strictly related to this feature but does get us closer to #5645